2023-09-18 17:44:45.420510: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'cudart64_100.dll'; dlerror: cudart64_100.dll not found
2023-09-18 17:44:45.420688: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
INFO flwr 2023-09-18 17:44:53,069 | app.py:151 | Starting Flower server, config: ServerConfig(num_rounds=10, round_timeout=None)
INFO flwr 2023-09-18 17:44:53,141 | app.py:172 | Flower ECE: gRPC server running (10 rounds), SSL is disabled
INFO flwr 2023-09-18 17:44:53,141 | server.py:86 | Initializing global parameters
INFO flwr 2023-09-18 17:44:53,141 | server.py:273 | Requesting initial parameters from one random client
INFO flwr 2023-09-18 17:44:53,230 | server.py:277 | Received initial parameters from one random client
INFO flwr 2023-09-18 17:44:53,230 | server.py:88 | Evaluating initial parameters
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:44:53,247 | server.py:95 | initial parameters (loss, other metrics): 0.6931471805599453, {'accuracy': 0.44333333333333336}
INFO flwr 2023-09-18 17:44:53,247 | server.py:101 | FL starting
DEBUG flwr 2023-09-18 17:44:53,390 | server.py:223 | fit_round 1: strategy sampled 2 clients (out of 3)
DEBUG flwr 2023-09-18 17:44:53,411 | server.py:237 | fit_round 1 received 2 results and 0 failures
WARNING flwr 2023-09-18 17:44:53,413 | fedavg.py:243 | No fit_metrics_aggregation_fn provided
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:44:53,420 | server.py:125 | fit progress: (1, 0.40023205378189514, {'accuracy': 0.82}, 0.17319209999999963)
DEBUG flwr 2023-09-18 17:44:53,421 | server.py:173 | evaluate_round 1: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:44:53,443 | server.py:187 | evaluate_round 1 received 3 results and 0 failures
WARNING flwr 2023-09-18 17:44:53,444 | fedavg.py:274 | No evaluate_metrics_aggregation_fn provided
DEBUG flwr 2023-09-18 17:44:53,444 | server.py:223 | fit_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:44:53,465 | server.py:237 | fit_round 2 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:44:53,476 | server.py:125 | fit progress: (2, 0.3987251592025144, {'accuracy': 0.81}, 0.22842879999999965)
DEBUG flwr 2023-09-18 17:44:53,476 | server.py:173 | evaluate_round 2: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 17:44:53,499 | server.py:187 | evaluate_round 2 received 4 results and 0 failures
DEBUG flwr 2023-09-18 17:44:53,500 | server.py:223 | fit_round 3: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 17:44:53,526 | server.py:237 | fit_round 3 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:44:53,537 | server.py:125 | fit progress: (3, 0.4012046912079904, {'accuracy': 0.8166666666666667}, 0.2896958000000005)
DEBUG flwr 2023-09-18 17:44:53,538 | server.py:173 | evaluate_round 3: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 17:44:53,558 | server.py:187 | evaluate_round 3 received 4 results and 0 failures
DEBUG flwr 2023-09-18 17:44:53,558 | server.py:223 | fit_round 4: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 17:44:53,581 | server.py:237 | fit_round 4 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:44:53,592 | server.py:125 | fit progress: (4, 0.40120142629759664, {'accuracy': 0.8166666666666667}, 0.3439703000000005)
DEBUG flwr 2023-09-18 17:44:53,592 | server.py:173 | evaluate_round 4: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 17:44:53,612 | server.py:187 | evaluate_round 4 received 4 results and 0 failures
DEBUG flwr 2023-09-18 17:44:53,612 | server.py:223 | fit_round 5: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:44:53,640 | server.py:237 | fit_round 5 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:44:53,652 | server.py:125 | fit progress: (5, 0.4018576861225093, {'accuracy': 0.8166666666666667}, 0.40433070000000093)
DEBUG flwr 2023-09-18 17:44:53,652 | server.py:173 | evaluate_round 5: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:44:53,673 | server.py:187 | evaluate_round 5 received 5 results and 0 failures
DEBUG flwr 2023-09-18 17:44:53,673 | server.py:223 | fit_round 6: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:44:53,706 | server.py:237 | fit_round 6 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:44:53,718 | server.py:125 | fit progress: (6, 0.40289831995972364, {'accuracy': 0.8233333333333334}, 0.47028670000000083)
DEBUG flwr 2023-09-18 17:44:53,718 | server.py:173 | evaluate_round 6: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:44:53,741 | server.py:187 | evaluate_round 6 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:44:53,741 | server.py:223 | fit_round 7: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:44:53,775 | server.py:237 | fit_round 7 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:44:53,788 | server.py:125 | fit progress: (7, 0.40292488870247106, {'accuracy': 0.8233333333333334}, 0.5399083000000005)
DEBUG flwr 2023-09-18 17:44:53,788 | server.py:173 | evaluate_round 7: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:44:53,810 | server.py:187 | evaluate_round 7 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:44:53,811 | server.py:223 | fit_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:44:53,850 | server.py:237 | fit_round 8 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:44:53,862 | server.py:125 | fit progress: (8, 0.4029234697955391, {'accuracy': 0.8233333333333334}, 0.6135029000000003)
DEBUG flwr 2023-09-18 17:44:53,862 | server.py:173 | evaluate_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:44:53,885 | server.py:187 | evaluate_round 8 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:44:53,885 | server.py:223 | fit_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:44:53,926 | server.py:237 | fit_round 9 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:44:53,938 | server.py:125 | fit progress: (9, 0.4029235906892315, {'accuracy': 0.8233333333333334}, 0.6905359999999998)
DEBUG flwr 2023-09-18 17:44:53,939 | server.py:173 | evaluate_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:44:53,967 | server.py:187 | evaluate_round 9 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:44:53,967 | server.py:223 | fit_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:44:54,004 | server.py:237 | fit_round 10 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:44:54,020 | server.py:125 | fit progress: (10, 0.40292358205797657, {'accuracy': 0.8233333333333334}, 0.7726915000000005)
DEBUG flwr 2023-09-18 17:44:54,020 | server.py:173 | evaluate_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:44:54,047 | server.py:187 | evaluate_round 10 received 6 results and 0 failures
INFO flwr 2023-09-18 17:44:54,048 | server.py:147 | FL finished in 0.8003754999999995
INFO flwr 2023-09-18 17:44:54,048 | app.py:218 | app_fit: losses_distributed [(1, 0.3877512253820896), (2, 0.4035768538713455), (3, 0.40146497984727225), (4, 0.40146538317203523), (5, 0.4075798676127479), (6, 0.4087129209722791), (7, 0.4087136285645621), (8, 0.40871365921837943), (9, 0.40871364772319796), (10, 0.40871366219861166)]
INFO flwr 2023-09-18 17:44:54,049 | app.py:219 | app_fit: metrics_distributed_fit {}
INFO flwr 2023-09-18 17:44:54,049 | app.py:220 | app_fit: metrics_distributed {}
INFO flwr 2023-09-18 17:44:54,049 | app.py:221 | app_fit: losses_centralized [(0, 0.6931471805599453), (1, 0.40023205378189514), (2, 0.3987251592025144), (3, 0.4012046912079904), (4, 0.40120142629759664), (5, 0.4018576861225093), (6, 0.40289831995972364), (7, 0.40292488870247106), (8, 0.4029234697955391), (9, 0.4029235906892315), (10, 0.40292358205797657)]
INFO flwr 2023-09-18 17:44:54,049 | app.py:222 | app_fit: metrics_centralized {'accuracy': [(0, 0.44333333333333336), (1, 0.82), (2, 0.81), (3, 0.8166666666666667), (4, 0.8166666666666667), (5, 0.8166666666666667), (6, 0.8233333333333334), (7, 0.8233333333333334), (8, 0.8233333333333334), (9, 0.8233333333333334), (10, 0.8233333333333334)]}
>>>>>>>>>>>>> setting initial parameters
[[0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0.]]
>>>>>>>>>>>>> evaluating on server-side
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.8196631 ,  0.93513549,  0.75436078,  0.51020884,  0.02777293,
         0.22209263, -0.09938505,  0.04138249]]), array([-0.06021571])]
[array([[-1.8196631 ,  0.93513549,  0.75436078,  0.51020884,  0.02777293,
         0.22209263, -0.09938505,  0.04138249]]), array([-0.06021571])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.05691677,  1.17268525,  0.76479373,  0.4957114 ,  0.14556986,
         0.10688754, -0.1143986 ,  0.0229796 ]]), array([0.019257])]
[array([[-2.05691677,  1.17268525,  0.76479373,  0.4957114 ,  0.14556986,
         0.10688754, -0.1143986 ,  0.0229796 ]]), array([0.019257])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.98677096,  1.07816838,  0.68214069,  0.46884574,  0.22403483,
         0.08572749,  0.00968052, -0.01290132]]), array([0.02214423])]
[array([[-1.98677096,  1.07816838,  0.68214069,  0.46884574,  0.22403483,
         0.08572749,  0.00968052, -0.01290132]]), array([0.02214423])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.98751344,  1.07951203,  0.68283309,  0.4691607 ,  0.22414152,
         0.08578771,  0.01112988, -0.01275425]]), array([0.01799273])]
[array([[-1.98751344,  1.07951203,  0.68283309,  0.4691607 ,  0.22414152,
         0.08578771,  0.01112988, -0.01275425]]), array([0.01799273])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.98291493,  1.04928319,  0.68301412,  0.4269406 ,  0.2158917 ,
         0.08872484,  0.00735046, -0.01492312]]), array([0.04404321])]
[array([[-1.98291493,  1.04928319,  0.68301412,  0.4269406 ,  0.2158917 ,
         0.08872484,  0.00735046, -0.01492312]]), array([0.04404321])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.98646453,  1.00795721,  0.70056213,  0.41010883,  0.19826661,
         0.09801192,  0.0434109 ,  0.01812759]]), array([0.03498675])]
[array([[-1.98646453,  1.00795721,  0.70056213,  0.41010883,  0.19826661,
         0.09801192,  0.0434109 ,  0.01812759]]), array([0.03498675])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.98631797,  1.00704115,  0.69957742,  0.40935242,  0.19764908,
         0.09742016,  0.04371199,  0.01789667]]), array([0.03973575])]
[array([[-1.98631797,  1.00704115,  0.69957742,  0.40935242,  0.19764908,
         0.09742016,  0.04371199,  0.01789667]]), array([0.03973575])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.98634845,  1.00707551,  0.69951501,  0.40934983,  0.1976533 ,
         0.0974095 ,  0.04369006,  0.0178547 ]]), array([0.0398386])]
[array([[-1.98634845,  1.00707551,  0.69951501,  0.40934983,  0.1976533 ,
         0.0974095 ,  0.04369006,  0.0178547 ]]), array([0.0398386])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.98634748,  1.0070744 ,  0.69952159,  0.40934968,  0.19765625,
         0.09741268,  0.04369176,  0.01785929]]), array([0.03982033])]
[array([[-1.98634748,  1.0070744 ,  0.69952159,  0.40934968,  0.19765625,
         0.09741268,  0.04369176,  0.01785929]]), array([0.03982033])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.98634747,  1.00707442,  0.69952108,  0.40934969,  0.197656  ,
         0.09741244,  0.04369166,  0.01785895]]), array([0.03982174])]
[array([[-1.98634747,  1.00707442,  0.69952108,  0.40934969,  0.197656  ,
         0.09741244,  0.04369166,  0.01785895]]), array([0.03982174])]
>>>>>>>>>>>>>run time:
0.5s
